docs/pages/repo/docs/troubleshooting.mdx

```diff
@@ -5,37 +5,152 @@ description: This guide aims to help you debug issues with your Turborepo builds
+directions even if you aren't planning to make a contribution.
+
+1. Clone the git repo from [`vercel/turbo`][source]
+1. `cd cli`
+1. Make any changes (for example, add more logging)
+1. Run `make`
+1. From _your_ project, use `/:path/:to/:turbo/target/debug/turbo` instead of global
+   turbo or the version of `turbo` installed in your project.
+
+## Common Pitfalls
+
+### The `.turbo` directory
+
+One of the [core concepts][2] behind Turbo is that when a declared input
+changes, the cached outputs for that task are invalidated. As part of running any task,
+Turborepo creates the following directories:
+
+- A `.turbo` at the root of your repo
+- A `.turbo` directory in each workspace if your project is a monorepo (e.g. `apps/my-app/.turbo/`)
+- A `turbo` directory in `node_modules/.cache`
 
-## I'm not seeing any cache hits
+Becuase the first two directories are not git-ignored by default, you may see an
+issue where you run the same task twice and get a cache missg, even though you
+didn't change anything, because the generated `.turbo` directories are getting included as
+the task _inputs_, and invalidating cache. To avoid this problem, you can limit your
+[`inputs` configuration][r-inputs-config], or add `.turbo/` to your `.gitignore` file. We recommend
+the latter.
+
+## Common Questions
+
+### I'm not seeing any cache hits
 
 - Is any source code being generated during the build that isn't checked into git?
-  - This would change the fingerprint Turborepo uses to store build outputs.
-- Are [cache outputs correctly specified](/repo/docs/core-concepts/caching#configuring-cache-outputs) in your Turborepo [pipeline](/repo/docs/core-concepts/monorepos/running-tasks#defining-a-pipeline)?
-  - Pipeline settings are not inherited or merged, so they need to be re-specified in [workspace-specific tasks](/repo/docs/core-concepts/monorepos/running-tasks#specific-workspace-tasks) (e.g. `web#build` does **not** inherit pipeline settings from `build`).
-- [Are relevant inlined environment variables accounted for?](/repo/docs/core-concepts/caching#altering-caching-based-on-environment-variables)
-  - To verify, run `turbo` in verbose mode by adding `-vvv` to `turbo run <task>` and look at which environment variables are included in the hashes.
 
-## I'm seeing cache hits, but my build is broken
+  This would change the fingerprint Turborepo uses to store build outputs.
+
+- Are [cache outputs correctly specified][d-config-outputs] in your Turborepo [pipeline][d-def-pipeline]?
+
+  Pipeline settings are not inherited or merged, so they need to be
+  re-specified in [workspace-specific tasks][d-workspace-tasks] (e.g. `web#build` does
+  **not** inherit pipeline settings from `build`).
+
+- [Are relevant inlined environment variables accounted for?][12]
+
+  To verify, run `turbo` in verbose mode by adding `-vvv` to
+  `turbo run <task>` and look at which environment variables are included in
```

Comment:
As we move more stuff into rust, -vvv is going to get very very very verbose (about 12 log lines for each grpc request, for example). We may want to reconsider what goes in debug vs trace.

Reply:
yeah agreed, i'm not sure what the levels are currently, but breaking them down into more levels makes sense to me.